### PR TITLE
[OSDOCS-7329]:Need Firewall prerequisites for OSD on GCP/[OSDOCS-12451]: Update "Customer Cloud Subscriptions on GCP" topic

### PR DESCRIPTION
**Pull request by mletalie**
Version(s):

4.17+
Issue:

https://issues.redhat.com/browse/OSDOCS-7329
https://issues.redhat.com/browse/OSDOCS-12451
Link to docs preview:

https://82600--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_planning/gcp-ccs.html#osd-gcp-psc-firewall-prerequisites_gcp-ccs
QE review:
- [x] QE has approved this change.

Additional information:

**QE REVIEWER  PLEASE NOTE**: SRE (SMEs) have determined no prerequisite is needed to include the allowlist urls for the the firewall.  Please confirm.
See https://redhat-internal.slack.com/archives/C065J9YCB34/p1727456181440069?thread_ts=1724823888.735349&cid=C065J9YCB34 for more context.

### CODE REVIEW

**Comment by mletalie** [comment]
Hello @lento-sun, may I get a QE review on this PR when you have a moment? Thanks!

**Comment by lento-sun** [comment]
Hi @mletalie I checked with the alllowlist doc (hope it's the correct doc): https://docs.google.com/document/d/1tMO3kkUri0EChNvtilPmtNWT-DX3d-00JU88Autq01o

And seems following urls not in the pr:
time-a-g.nist.gov
time-a-wwv.nist.gov
time-a-b.nist.gov

I saw some discussion in the doc's comments, but seems no conclusion. Could you pls help to confirm if they are not required in the list, thx

**Review by michaelryanmcneill — CHANGES_REQUESTED** [review]
Just a few changes!

File: modules/osd-gcp-psc-firewall-prerequisites.adoc [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+ `oauth2.googleapis.com`
+
+| 443
+| Required to access GCP services and resources. Review link:https://cloud.google.com/endpoints/docs[Cloud Endpoints] in the GCP documentation to determine the endpoints to allow for your APIs.
+|===
+
+[NOTE]
+====
+Google APIs can be consumed using the link:https://cloud.google.com/vpc-service-controls/docs/restricted-vip-services[Virtual IP (VIP) with Private Google Access], with the exception of the Service Usage API (serviceusage.googleapis.com). To circumvent this, you must expose the Service Usage API by configuring link:https://cloud.google.com/vpc/docs/configure-private-google-access#domain-options[Private Google Access].
```
[the remark this answers is not in the document]
Comment (mletalie) [reply modules/osd-gcp-psc-firewall-prerequisites.adoc:250]: Done

File: modules/osd-gcp-psc-firewall-prerequisites.adoc [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+|`http-inputs-osdsecuritylogs.splunkcloud.com`
+|443
+|Required. Used by the `splunk-forwarder-operator` as a logging forwarding endpoint to be used by Red{nbsp}Hat SRE for log-based alerting.
+
+|`sftp.access.redhat.com` (Recommended)
+|22
+|The SFTP server used by `must-gather-operator` to upload diagnostic logs to help troubleshoot issues with the cluster.
+|===
+
+. Allowlist the following URLs for optional third-party content:
```
[the remark this answers is not in the document]
Comment (mletalie) [reply modules/osd-gcp-psc-firewall-prerequisites.adoc]: Removed

File: modules/osd-gcp-psc-firewall-prerequisites.adoc [hunk 3]
```diff
@@ -0,0 +1,232 @@
+// Module included in the following assemblies:
+//
+// * osd_planning/gcp-ccs.adoc
+
+
```
Comment (AedinC) [inline modules/osd-gcp-psc-firewall-prerequisites.adoc:5]:
```suggestion
:_mod-docs-content-type: PROCEDURE
```

**Review by AedinC — COMMENTED** [review]
Looks great @mletalie, just a few very minor suggestions.

**Comment by mletalie** [comment]
> Just a few changes!

Updated, thanks!

**Comment by mletalie** [comment]
QE and peer reviewed.

File: modules/osd-gcp-psc-firewall-prerequisites.adoc [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+. Add the following {product-title} URLs to an allowlist:
++
+[cols="6,1,6",options="header"]
+|===
+|Domain | Port | Function
+
+|`mirror.openshift.com`
+|443
+|Used to access mirrored installation content and images. This site is also a source of release image signatures, although the Cluster Version Operator (CVO) needs only a single functioning source.
```
Comment (michaelryanmcneill) [inline modules/osd-gcp-psc-firewall-prerequisites.adoc:127]: Change requested: 
> Used to access mirrored installation content and images. This site is also a source of release image signatures.

File: modules/osd-gcp-psc-firewall-prerequisites.adoc [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
++
+[cols="6,1,6",options="header"]
+|===
+|Domain | Port | Function
+
+| `accounts.google.com`
+| 443
+| Used to access your GCP account.
+
+|`* .googleapis.com`
```
Comment (michaelryanmcneill) [inline modules/osd-gcp-psc-firewall-prerequisites.adoc:209]: Requested change:
> `*.googleapis.com`

File: osd_planning/gcp-ccs.adoc [hunk 6]
```diff
@@ -15,8 +15,11 @@ include::modules/ccs-gcp-customer-procedure.adoc[leveloffset=+1]
...
 include::modules/ccs-gcp-provisioned.adoc[leveloffset=+1]
 include::modules/gcp-limits.adoc[leveloffset=+1]
+include::modules/osd-gcp-psc-firewall-prerequisites.adoc[leveloffset=+1]
 
 [id="additional-resources_{context}"]
 == Additional resources
 
-* xref:../security/rh-required-whitelisted-IP-addresses-for-sre-access.adoc#rh-required-whitelisted-IP-addresses-for-sre-access[Required allowlist IP addresses for SRE access]
\ No newline at end of file
+* xref:../security/rh-required-whitelisted-IP-addresses-for-sre-access.adoc#rh-required-whitelisted-IP-addresses-for-sre-access[Required allowlist IP addresses for SRE access]
```
Comment (michaelryanmcneill) [inline osd_planning/gcp-ccs.adoc:23]: This should be removed from this page as it is not applicable. I'm not actually sure the purpose of that page at all, as it also doesn't apply to OSD on AWS...

**Review by michaelryanmcneill — CHANGES_REQUESTED** [review]
A few additional changes, thanks @mletalie for all your work!

File: osd_planning/gcp-ccs.adoc (hunk 6 above)
Comment (mletalie) [reply osd_planning/gcp-ccs.adoc:23]: Hi @michaelryanmcneill, Did some research and came up with this doc sticket: https://issues.redhat.com/browse/OSDOCS-11353
I can/will remove link from page but wait until you get a chance to look at the docs ticket to verify if you believe this was added erroneously before making additional changes. If we do decide to remove topic entirely, will need to open an additional PR for that. thx. Thanks.

Comment (michaelryanmcneill) [reply osd_planning/gcp-ccs.adoc:23]: Opened a thread in Slack to discuss this further, will circle back shortly!

Comment (mletalie) [reply osd_planning/gcp-ccs.adoc:23]: Created a docs ticket to address this issue once we determine best path forward. Thanks! https://issues.redhat.com/browse/OSDOCS-12438

File: modules/osd-gcp-psc-firewall-prerequisites.adoc [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+|===
+|Domain | Port | Function
+
+| `accounts.google.com`
+| 443
+| Used to access your GCP account.
+
+|`*.googleapis.com`
+
+OR
```
Comment (michaelryanmcneill) [inline modules/osd-gcp-psc-firewall-prerequisites.adoc:211]: We need to add a two other ones to this list, sorry for the miss:
```
dns.googleapis.com
iamcredentials.googleapis.com
```

Comment (mletalie) [reply modules/osd-gcp-psc-firewall-prerequisites.adoc:211]: Got it, thanks for catching. Will update today.

**Pull request merged**